core/core.go

```diff
@@ -510,7 +526,29 @@ func (t *Textile) threadInterceptor() grpc.UnaryServerInterceptor {
 		// This needs to happen before the request is handled in case there's a conflict
 		// with the owner and thread name.
 		if newID.Defined() {
-			if _, err := t.co.Threads.Create(ctx, newID, owner); err != nil {
+			if _, err := t.co.Threads.Create(ctx, newID, owner, isDB); err != nil {
+				return nil, err
+			}
+		}
+
+		// Track the thread ID marked for deletion.
+		var deleteID thread.ID
+		switch method {
+		case "/threads.pb.API/DeleteDB":
+			deleteID, err = thread.Cast(req.(*dbpb.DeleteDBRequest).DbID)
+			if err != nil {
+				return nil, err
+			}
+			keys, err := t.co.IPNSKeys.ListByThreadID(ctx, deleteID)
+			if err != nil {
+				return nil, err
+			}
+			if len(keys) != 0 {
+				return nil, status.Error(codes.FailedPrecondition, "DB not empty. Delete buckets first.")
```

Comment:
A condition I came up with that feels reasonable: You cannot delete a database without first deleting all buckets, _unless_ you're destroying your entire account.